### DISCUSSION TITLE
[Run issue]: Error in running example for post-instalation \\ libG4Tree.so: cannot open shared object file: No such file or directory

### DISCUSSION THREAD

**Discussion by ghost** [OpenTOPAS/OpenTOPAS#40] [Run-Problem] [answered]
### Contact Details

cecilianovawijaya01@gmail.com

### What was the issue?

Hello! 
I have finished installing OpenTOPAS but I experience error while trying to run the example

I have created the OpenTOPAS shell script and exported to their path.
![Selection_011](https://github.com/OpenTOPAS/OpenTOPAS/assets/152828218/e6b08c6a-78e7-4434-8084-12b72c96a58d)

Here is the shell script that was created under name topas in the ~/shellScripts folder
```
export QT_QPA_PLATFORM_PLUGIN_PATH=$HOME/software/OpenTopas/OpenTOPAS-install/Frameworks
export TOPAS_G4_DATA_DIR=$HOME/G4Data
export LD_LIBRARY_PATH=$HOME/software/OpenTopas/OpenTOPAS-install/lib:$LD_LIBRARY_PATH
export LD_LIBRARY_PATH=$HOME/software/geant4/geant4-v11.1.3-install/lib:$LD_LIBRARY_PATH
$HOME/software/OpenTopas/OpenTOPAS-install/bin/topas $1
```

The path is already in accordance with the OpenTOPAS and Geant4 folder in my computer. This is my .bashrc after echo-ing the path
![Selection_010](https://github.com/OpenTOPAS/OpenTOPAS/assets/152828218/a58c4591-0b2a-4702-8f3b-56bb78aaa1e0)

This is the error message that I received
```
(base) cecilia@cecilia-X570-AORUS-ELITE:~$ source ~/.bashrc
(base) cecilia@cecilia-X570-AORUS-ELITE:~$ cd $HOME/software/OpenTopas/OpenTOPAS-install/examples/Basic
(base) cecilia@cecilia-X570-AORUS-ELITE:~/software/OpenTopas/OpenTOPAS-install/examples/Basic$ ~/software/OpenTopas/OpenTOPAS-install/bin/topas QtShapeTest.txt 
/home/cecilia/software/OpenTopas/OpenTOPAS-install/bin/topas: error while loading shared libraries: libG4Tree.so: cannot open shared object file: No such file or directory
```

I have checked the Library Path and the file libG4Tree.so exist in that directory
![Selection_009](https://github.com/OpenTOPAS/OpenTOPAS/assets/152828218/3cf383f9-ac75-4b4f-97cb-7d173648dd8a)

Can anyone please help me how to solve this? 

[Ps: I am not sure which version of TOPAS I am running since I only follow this [instruction](https://github.com/OpenTOPAS/OpenTOPAS/blob/master/OpenTOPAS_quickStart_Debian.md)

Thank you for your attention!

### TOPAS category

Geometry

### Version

3.7

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Attestation

- [X] I testify that I have fully read the bug reporting guideline beforehand.

**Comment by icmeyer** [comment, marked as answer]
Hi,
Most of your installation looks correct. Right now, the environment has to be set up manually for each TOPAS run using the shell script. So in order to run TOPAS you will need to change `~/software/OpenTopas/OpenTOPAS-install/bin/topas QtShapeTest.txt` to just
`topas QtShapeTest.txt`
The TOPAS executable is wrapped by the shell script, which you can also reference it directly:
`~/shellScripts/topas QtShapeTest.txt`

Let me know if that still leads to bugs!

**Reply by ghost** [reply]
It works well now!  I have finished installing the OpenTOPAS

Thank you so much for your answer^^

**Reply by flakesflakes** [reply]
Hi, I am having the same issue here but using either 
~/Applications/TOPAS/OpenTOPAS-install/bin/topas QtShapeTest.txt or topas QtShapeTest.txt (listing the environment without the script) or topas QtShapeTest.txt (with the bash) gives me the same error.

![image](https://github.com/user-attachments/assets/0cf51603-6667-47a7-b4b3-a3808ae70116)

![image](https://github.com/user-attachments/assets/1b51430a-88a0-4720-ad47-38a459e6f0b1)

My installation in a cluster environment looks the same besides the name of the folder where the file libG4Tree.so is located, here's at lib64 instead of lib
![image](https://github.com/user-attachments/assets/b3bef37d-55c8-406f-8d73-d4da9dffc07b)

I also followed the same instructions.
Could anyone help with that?

**Reply by ghost** [reply]
Hello!

Maybe instead of 
`export LD_LIBRARY_PATH=$HOME/Applications/GEANT4/geant4-install/lib:LD_LIBRARY_PATH`, 
try 
`export LD_LIBRARY_PATH=$HOME/Applications/GEANT4/geant4-install/lib64:LD_LIBRARY_PATH`.

If that works, you might also need to update your shell script file.

I’m not an expert, so I’m not 100% sure haha! But it might be worth a try.

Best,
Cecilia
